src/mlpack/bindings/julia/get_printable_type_impl.hpp

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file get_printable_type_impl.hpp
+ * @author Ryan Curtin
+ *
+ * Get the printable type of a parameter.  This type is not the C++ type but
+ * instead the Julia type that a user would use.
+ */
+#ifndef MLPACK_BINDINGS_JULIA_GET_PRINTABLE_TYPE_IMPL_HPP
+#define MLPACK_BINDINGS_JULIA_GET_PRINTABLE_TYPE_IMPL_HPP
+
+#include "get_printable_type.hpp"
+
+namespace mlpack {
+namespace bindings {
+namespace julia {
+
+/**
+ * Return a string representing the command-line type of an option.
+ */
+template<typename T>
+std::string GetPrintableType(
+    const util::ParamData& data,
+    const typename boost::disable_if<arma::is_arma_type<T>>::type*,
+    const typename boost::disable_if<util::IsStdVector<T>>::type*,
+    const typename boost::disable_if<data::HasSerialize<T>>::type*,
+    const typename boost::disable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  if (std::is_same<T, bool>::value)
+    return "Bool";
+  else if (std::is_same<T, int>::value)
+    return "Int";
+  else if (std::is_same<T, double>::value)
+    return "Float64";
+  else if (std::is_same<T, std::string>::value)
+    return "String";
+  else
+    throw std::invalid_argument("unknown parameter type" + data.cppType);
```

Comment:
Should there be a space right after type?

Reply:
Yes, nice catch; fixed in eee1773.